test/integration/smoke/test_volumes.py

```diff
@@ -383,29 +367,22 @@ def setUp(self):
                     available")
 
     def tearDown(self):
-        #Clean up, terminate the created volumes
-        if self.attached:
-            self.virtual_machine.get_ssh_client(reconnect = True)
-
-            self.virtual_machine.detach_volume(self.apiClient, self.volume)
-
         if self.virtual_machine.hypervisor == "KVM":
             self.virtual_machine.stop(self.apiClient)
+            if self.attached:
+                self.virtual_machine.detach_volume(self.apiClient, self.volume)
             self.virtual_machine.start(self.apiClient)
-            self.virtual_machine.get_ssh_client(reconnect = True)
-
+            self.virtual_machine.get_ssh_client(reconnect=True)
```

Comment:
```suggestion
            try:
                self.virtual_machine.get_ssh_client(reconnect=True)
            except Exception as err:
                self.fail("SSH failed for Virtual machine: %s due to %s" %
                      self.virtual_machine.ipaddress, err)
```
To catch it if it fails here